handlers/new-product-api/src/main/scala/com/gu/newproduct/api/addsubscription/Handler.scala

```diff
@@ -38,10 +38,23 @@ object Handler extends Logging {
     ApiGatewayHandler(LambdaIO(inputStream, outputStream, context)) {
       Steps.operationForEffects(RawEffects.response, RawEffects.stage, GetFromS3.fetchString, SqsAsync.send(SqsAsync.buildClient), RawEffects.now)
     }
+
+  def main(args: Array[String]): Unit = {
```

Comment:
not sure what the something is - would this just wire up the dependencies and waiting for an input stream/request body?
```suggestion
  // run this locally to something?
  def main(args: Array[String]): Unit = {
```

Reply:
I'm confused as to what is this new main method is supposed to do. It looks like it only wires the code but it never makes it run.

To run this locally we have [this code](https://github.com/guardian/support-service-lambdas/blob/68c025ada78ab38c8bbf06526194fa63b083ace4/handlers/new-product-api/src/test/scala/manualTest/AddSubscriptionManualTest.scala#L9) that pretty much calls the handler but it passes a hardcoded request json in the inputstream and it prints out the response from the outputstream

Reply:
Yes you're right this was an attempt to run the logic and verify that it ran without error, but it really didn't work very well since it needs input of some sort and I haven't figured out how to do that yet. Thanks for the pointer towards AddSubscriptionManualTest as an alternative